akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpOriginRange.java

```diff
@@ -10,10 +10,10 @@
 /**
  * @see HttpOriginRanges for convenience access to often used values.
  */
-public abstract class HttpOriginRange {
-    public abstract boolean matches(HttpOrigin origin);
+public interface HttpOriginRange {
```

Comment:
We should document it more explicitly in  http://doc.akka.io/docs/akka/snapshot/common/binary-compatibility-rules.html but there is a group of classes which are "public API, but users should not extend it", under such rules (applying to the HTTP model) we can do such changes.